sklearn/model_selection/_plot.py

```diff
@@ -648,11 +664,15 @@ def plot(
             `scikit-learn`.
 
         score_name : str, default=None
-            The name of the score used to decorate the y-axis of the plot. If
-            `None` and `scoring` is provided, the same will be inferred from it
-            by removing `"neg_"` and replacing underscore by spaces. If
-            `scoring is None`, we return the default value `"Negative score"`
-            if `negate_score` is `True` and `"Score"` otherwise.
+            The name of the score used to decorate the y-axis of the plot.
+            It will override the name inferred from the `scoring` parameter.
+            If `score` is `None`, we use `"Score"` if `negate_score` is `False` and
+            `"Negative score"` otherwise.
+            If `scoring` is a string or a callable, we infer the name:
+
+            - we replace `_` by spaces and capitalize the first letter;
+            - we remove `neg_` and replace it by `"Negative"` if `negate_score` is
+            `True` or just remove it otherwise.
```

Comment:
```suggestion
            - we remove `neg_` and replace it by `"Negative"` if `negate_score` is
            `False` or just remove it otherwise.
```